### PR TITLE
Allow custom DLL names

### PR DESCRIPTION
**Pull request by ethomson**
Like in the pre-NuGet native binary days, we require a way to have a custom libgit2 filename, for example `git2-visualstudio.dll`.  This updates the build and update scripts to include support for this, as well as the ability to then output a custom NuGet package suffix so that we do not confuse *our* binary packages with the official ones.

This also adds a `README.md` describing how to work with this system.

### CODE REVIEW

**Comment by ethomson** [comment]
Oh, doh, @nulltoken - I didn't realize that you had a `README` in another PR.  Let me know if you want me to drop that commit and we can figure something else out.

**Comment by bording** [comment]
Since the changes in libgit2/libgit2sharp#1157 require `libgit2_filename.txt` to exist in the package, this file also needs to be commited with the standard naming scheme.

Other than that, I think this looks good. I don't see anything in this that would break on the mono side of things when using a custom name. Did you by any chance test that to confirm? If not, I can double check that.

**Comment by Therzok** [comment]
@bording as long as the dllimport's name is regenerated, nothing will break on mono side.

**Comment by bording** [comment]
@Therzok My comment was more about making sure the custom name was actually propagated out to the appropriate places for mono to see it, for example the LibGit2Sharp.dll.config file, and it looks like that is all being handled correctly.

**Comment by ethomson** [comment]
> Since the changes in libgit2/libgit2sharp#1157 require libgit2_filename.txt to exist in the package, this file also needs to be commited with the standard naming scheme.

Makes sense.  Fixed that up.  Thanks.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+[Libgit2Sharp](https://github.com/libgit2/libgit2sharp) is a managed
+wrapper around [libgit2](https://github.com/libgit2/libgit2), and as
+such requires compilation of libgit2 for your platform.  LibGit2Sharp
+makes this easy by distributing NuGet packages that include precompiled
+versions of these native binaries.
+
+If you need to build your own native binaries for some reason, you can
+do so easily with the scripts in this repository:
+
+1. Clone the LibGit2Sharp repository.  Do so recursively to ensure that
```
Comment (nulltoken) [inline README.md]: Clone the LibGit2Sharp.NativeBinaries repository

File: README.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   `build.libgit2.sh`
+
+4. Create the NuGet package from the built binaries.  You will need to
+   specify the version number of the resultant NuGet package that you
+   want to generate.  Note that you may wish to provide a suffix to
+   disambiguate your custom package from the official, published NuGet
+   packages.  For example, if you are building a product called
+   `fooproduct` then that may be a helpful suffix.
+
+    To build a NuGet package at version `1.2.3-foo`:
```
Comment (nulltoken) [inline README.md]: `-foo` will still be seen as a pre-release package in the eyes of NuGet. However, I cannot see a better approach.

Comment (ethomson) [inline README.md]: Yeah, I can't either.  Would you like me to make a note of that in this file for clarification?

**Comment by nulltoken** [comment]
> Oh, doh, @nulltoken - I didn't realize that you had a README in another PR. Let me know if you want me to drop that commit and we can figure something else out.

Are you kidding me? This is an amazing README! Of course, you keep it.

However, I wouldn't mind some :eyes: on #5 :wink:

File: README.md (hunk 2 above)
Comment (nulltoken) [inline README.md]: >Would you like me to make a note of that in this file for clarification?

That'd be great

File: buildpackage.ps1 [hunk 3]
```diff
@@ -38,4 +38,4 @@ if ( -Not (Test-Path $linuxDirectory\*.so) )
     Set-Content $linuxDirectory\addbinaries.here $null
 }
 
-Nuget.exe Pack nuget.package\NativeBinaries.nuspec -Version $version$versionSuffix -NoPackageAnalysis
+.\Nuget.exe Pack nuget.package\NativeBinaries.nuspec -Version $version$versionSuffix -NoPackageAnalysis
```
Comment (nulltoken) [inline buildpackage.ps1]: **Disclaimer:** I'm not a PowerShell wizard. But doesn't the dot-slash notation notation means *"here"*? Would that be the case, how can we ensure that there's always a NuGet next to this file?

Comment (nulltoken) [inline buildpackage.ps1]: Drop this. I forgot that we actually embed NuGet.exe.

File: README.md (hunk 2 above)
Comment (ethomson) [inline README.md]: Added some notes in the paragraph below.

**Comment by ethomson** [comment]
@nulltoken I made some changes, but I didn't broadly include #5 yet.  There's some very good stuff in there and we should merge the two documents for sure, but I didn't do that as part of this PR.

**Comment by nulltoken** [comment]
@ethomson Neat!

**Pull request merged**